### DISCUSSION TITLE
4.7.x .NET Framework SDK Docker images no longer published

### DISCUSSION THREAD

**Discussion by mthalman** [microsoft/dotnet-framework-docker#920] [Announcements]
# 4.7.x .NET Framework SDK Docker images no longer published

The [4.7.x .NET Framework SDK Docker images](https://hub.docker.com/_/microsoft-dotnet-framework-sdk) will no longer be published after Oct. 8, 2019.  The 4.8 image tags should be used as a replacement.

## Details
The main reason for this was due to the incompatibility of the 4.7.1 SDK image with the Visual Studio 16.3 Build Tools (see #423).  It's always been a goal to maintain consistency of the Build Tools versions between all SDK images.  Since the 4.7.1 SDK image could not be upgraded to use 16.3 Build Tools, it meant that publishing for that image needed to be discontinued.  Based on that, the 4.7.2 SDK image was also included as part of this decision to have all consumers uptake the most current version, 4.8.

Any references to the following SDK tags should be updated to reference `4.8` instead:
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.1`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.1-windowsservercore-ltsc2016`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.1-<yyyymmdd>-windowsservercore-ltsc2016`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.2`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.2-windowsservercore-ltsc2016`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.2-<yyyymmdd>-windowsservercore-ltsc2016`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.2-windowsservercore-1803`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.2-<yyyymmdd>-windowsservercore-1803`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.2-windowsservercore-ltsc2019`
* `mcr.microsoft.com/dotnet/framework/sdk:4.7.2-<yyyymmdd>-windowsservercore-ltsc2019`

If you haven't migrated to using images from [MCR](https://github.com/dotnet/announcements/issues/101) and are still using the .NET Framework repos on Docker Hub, you will need to update your tag references to MCR.  This is because the 4.8 images only exist on MCR.  These are the following Docker Hub tags that are affected and should be updated to reference the [4.8 tags at the MCR location](https://hub.docker.com/_/microsoft-dotnet-framework-sdk):
* `microsoft/dotnet-framework:4.7.1-sdk`
* `microsoft/dotnet-framework:4.7.1-sdk-windowsservercore-ltsc2016`
* `microsoft/dotnet-framework:4.7.2-sdk`
* `microsoft/dotnet-framework:4.7.2-sdk-windowsservercore-ltsc2016`
* `microsoft/dotnet-framework:4.7.2-sdk-windowsservercore-1803`
* `microsoft/dotnet-framework:4.7.2-sdk-windowsservercore-ltsc2019`

**Comment by McDoit** [comment]
Is there any normal reason for the short notice, or should we expect an incoming CVE and Zero Day on this?

**Comment by mthalman** [comment]
@McDoit - The announcement has been updated to include extra information about why this change was made, including a link to the issue that prompted it, https://github.com/microsoft/dotnet-framework-docker/issues/423.
